src/Factory.php

```diff
@@ -389,20 +398,27 @@ private static function normalizeObject(object $object): object
         }
     }
 
-    private function normalizeCollection(?string $relationName, FactoryCollection $collection): array
+    private function normalizeCollection(string $relationName, FactoryCollection $collection): array
```

Comment:
something I've noticed here: if we replace this whole method with `return $collection->all();` the tests are not failing :shrug: 

so we'll need to understand what's really happening here, why (if?) it is needed, and how we could create a failing test case for this.